File: src/main/java/tools/important/tankslua/gui/screen/ScreenOptionsLua.java
package tools.important.tankslua.gui.screen;

import tanks.Drawing;
import tanks.Game;
import tanks.gui.Button;
import tanks.gui.screen.Screen;
import tanks.gui.screen.ScreenOptions;
import tools.important.tankslua.TanksLua;
import tools.important.tankslua.gui.ToggleOptionButton;

public class ScreenOptionsLua extends Screen {
    public Button backButton;
    public Button extensionListButton;

    public Button enableLevelScriptsButton;
    public Button enableEvalBoxButton;

    public ScreenOptionsLua() {
        backButton = new Button(centerX, centerY + objYSpace * 3.5, objWidth, objHeight, "Back", () -> Game.screen = new ScreenOptions());

        enableLevelScriptsButton = new ToggleOptionButton(centerX, centerY-objHeight*3, objWidth, objHeight, "Level scripts",
                () -> TanksLua.tanksLua.options.put("enableLevelScripts", true),
                () -> TanksLua.tanksLua.options.put("enableLevelScripts", false),
                (boolean)TanksLua.tanksLua.options.get("enableLevelScripts"),
                "Whether or not the extension should run scripts for specific levels------Please note that turning this off may cause problems with some levels---that depend on scripts to function!");
        enableEvalBoxButton = new ToggleOptionButton(centerX, centerY-objHeight*1.7, objWidth, objHeight, "Evaluate Box",
                () -> TanksLua.tanksLua.options.put("enableEvalBox", true),
                () -> TanksLua.tanksLua.options.put("enableEvalBox", false),
                (boolean) TanksLua.tanksLua.options.get("enableEvalBox"), "If the extension should show a box at the top of the screen---that lets you evaluate Lua code at will.------Note that it may block off certain UI elements sometimes,---so you shouldn't leave it on all the time!");


        extensionListButton = new Button(centerX, centerY+objHeight*4, objWidth, objHeight, "Loaded extension list", () -> Game.screen = new ScreenOptionsLuaExtensionList());
    }

    @Override
    public void update() {
        backButton.update();

        // ordering doesn't matter too much here
        enableLevelScriptsButton.update();
        enableEvalBoxButton.update();

        extensionListButton.update();
    }

    @Override
    public void draw() {
        drawDefaultBackground();
        backButton.draw();

        // make sure this ordering goes from lowest in ui to highest in ui
        enableEvalBoxButton.draw();
        enableLevelScriptsButton.draw();
        extensionListButton.draw();

        Drawing.drawing.setColor(0,0,0);
        Drawing.drawing.setInterfaceFontSize(titleSize);
        Drawing.drawing.displayInterfaceText(centerX, centerY - objYSpace * 3.5, "Lua Options");
    }
}

File: src/tools/important/tankslua/screen/ScreenOptionsLua.kt
package tools.important.tankslua.screen

import tanks.Drawing
import tanks.Game
import tanks.gui.Button
import tanks.gui.screen.Screen
import tanks.gui.screen.ScreenOptions
import tools.important.tankslua.TanksLua
import tools.important.tankslua.gui.ToggleButton
import tools.important.tankslua.openFileManagerTo
import tools.important.tankslua.tanksLuaDir
import tools.important.tankslua.verifyDirectoryStructure

class ScreenOptionsLua : Screen() {
    init {
        music = "menu_options.ogg"
        musicID = "menu"
    }

    val backButton = Button(
        centerX,
        centerY + objYSpace * 3.5,
        objWidth,
        objHeight,
        "Back"
    ) {
        Game.screen = ScreenOptions()
        TanksLua.options.save()
    }

    val toggleEvalBox = ToggleButton(
        centerX - objXSpace/2,
        centerY - objYSpace*0.5,
        objWidth,
        objHeight,
        "Eval box",
        { b -> TanksLua.options.evalBoxEnabled = b },
        TanksLua.options.evalBoxEnabled
    )

    val enableLevelScripts = ToggleButton(
        centerX - objXSpace/2,
        centerY - objYSpace*1.5,
        objWidth,
        objHeight,
        "Level scripts",
        { b -> TanksLua.options.levelScriptsEnabled = b },
        TanksLua.options.levelScriptsEnabled
    )

    val gotoLuaExtensions = Button(
        centerX + objXSpace/2,
        centerY - objYSpace*1.5,
        objWidth,
        objHeight,
        "Lua Extensions",
    ) {
        Game.screen = ScreenLuaExtensions()
    }

    val tanksluaFiles = Button(
        centerX + objXSpace/2,
        centerY - objYSpace*0.5,
        objWidth,
        objHeight,
        "   Open TanksLua folder", // The text will overlap with the folder icon without the spaces at the beginning
    ) {
        verifyDirectoryStructure()
        openFileManagerTo(tanksLuaDir)
    } .apply {
        image = "icons/folder.png"
        imageSizeX = objHeight * 0.75
        imageSizeY = objHeight * 0.75
        imageXOffset = -objWidth / 2.0 + imageSizeX
    }

    override fun update() {
        backButton.update()
        toggleEvalBox.update()
        gotoLuaExtensions.update()
        tanksluaFiles.update()
        enableLevelScripts.update()
    }

    override fun draw() {
        drawDefaultBackground()

        val drawing = Drawing.drawing!!

        drawing.setInterfaceFontSize(titleSize)
        drawing.setColor(0.0, 0.0, 0.0)

        drawing.displayInterfaceText(centerX, centerY - objYSpace * 3.5, "TanksLua Options")

        drawing.setInterfaceFontSize(titleSize/2)
        drawing.drawInterfaceText(centerX, centerY - objYSpace * 3.0, TanksLua.VERSION)

        backButton.draw()
        toggleEvalBox.draw()
        gotoLuaExtensions.draw()
        tanksluaFiles.draw()
        enableLevelScripts.draw()
    }
}